jadx-core/src/main/java/jadx/core/utils/files/JavaToDex.java

```diff
@@ -31,14 +31,16 @@ public DxArgs(DxContext context, String dexFile, String[] input) {
 
 	private String dxErrors;
 
-	public byte[] convert(String javaFile) throws JadxException {
+	public byte[] convert(String jarFile) throws JadxException {
 		try (ByteArrayOutputStream out = new ByteArrayOutputStream();
 		     ByteArrayOutputStream errOut = new ByteArrayOutputStream()) {
 			DxContext context = new DxContext(out, errOut);
-			DxArgs args = new DxArgs(context, "-", new String[]{javaFile});
+			DxArgs args = new DxArgs(context, "-", new String[]{jarFile});
 			int result = (new Main(context)).runDx(args);
 			dxErrors = errOut.toString(CHARSET_NAME);
 			if (result != 0) {
+				System.out.println(out);
+				System.err.println(errOut);
```

Comment:
I don't think this is a good solution. Yes, this is help in development, but users will don't see this anyway.
I already tried to process errors (you can see `dxErrors` field) and output them in exception and logs. Sometimes it works like in #390 (`dx message:`).
Also `out` can contains binary data.

Reply:
ok, thanks.